modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportReindexAction.java

```diff
@@ -82,15 +104,42 @@ protected void doExecute(ReindexRequest request, ActionListener<BulkIndexByScrol
         throw new UnsupportedOperationException("task required");
     }
 
+    private void checkRemoteWhitelist(RemoteInfo remoteInfo) {
+        TransportAddress publishAddress = null;
+        if (httpTransport != null && httpTransport.boundAddress() != null) {
+            publishAddress = httpTransport.boundAddress().publishAddress();
+        }
+        checkRemoteWhitelist(remoteWhitelist, remoteInfo, publishAddress);
+    }
+
+    static void checkRemoteWhitelist(Set<String> whitelist, RemoteInfo remoteInfo, TransportAddress publishAddress) {
+        if (remoteInfo == null) return;
+        if (whitelist.contains(remoteInfo.getHost())) return;
+        /*
+         * For testing we support the key "myself" to allow connecting to the local node. We can't just change the setting to include the
+         * local node because it is intentionally not a dynamic setting for security purposes. We can't use something like "localhost:9200"
+         * because we don't know up front which port we'll get because the tests bind to port 0. Instead we try to resolve it here, taking
+         * "myself" to mean "my published http address".
+         */
+        if (whitelist.contains("myself") && publishAddress != null && publishAddress.toString().equals(remoteInfo.getHost())) {
```

Comment:
I'm not super tied to the name. In fact I wish I didn't have to implement it at all but tests don't work without it because they bind to 0 like the comment says. Because it is a whitelist I think people should be super explicit with it. So I'd really prefer not to support all the `network.host` settings.